plugin/metrics/handler.go

```diff
@@ -24,11 +24,15 @@ func (m *Metrics) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg
 
 	// Record response to get status code and size of the reply.
 	rw := dnstest.NewRecorder(w)
-	status, err := plugin.NextOrFailure(m.Name(), m.Next, ctx, rw, r)
+	rcodeFromNext, err := plugin.NextOrFailure(m.Name(), m.Next, ctx, rw, r)
```

Comment:
plugins don't return an rcode, it's code to see if the server needs to write something to the client. So it next to useless for the purpose you want to use it for.